lib/pf/radius.pm

```diff
@@ -232,16 +232,11 @@ sub authorize {
     $args->{'autoreg'} = 0;
     # should we auto-register? let's ask the VLAN object
     my ( $status, $status_msg );
-    if ($role_obj->shouldAutoRegister($args)) {
+    $do_auto_reg = $role_obj->shouldAutoRegister($args);
+    if ($do_auto_reg) {
         $args->{'autoreg'} = 1;
-        # automatic registration
-        my %autoreg_node_defaults = $role_obj->getNodeInfoForAutoReg($args);
-        $args->{'node_info'} = merge($args->{'node_info'}, \%autoreg_node_defaults);
-        $logger->debug("[$mac] auto-registering node");
-        ( $status, $status_msg ) = pf::node::node_register($mac, $autoreg_node_defaults{'pid'}, %autoreg_node_defaults);
-        if (!$status) {
-            $logger->error("auto-registration of node failed");
-        }
+        $node_obj->autoreg("yes");
+        %autoreg_node_defaults = $role_obj->getNodeInfoForAutoReg($args);
```

Comment:
Correct me if I'm wrong but the node attributes / node info will not be modified during the autoreg process here but only at the end of the request.

That means the changes done to the node for auto-registration won't apply (status, role, ...) until the end of the request which means the node will not be auto-registered in this request and thus won't have the right RADIUS attributes sent back

From what I can see, the autoreg process in the cleanup should be done here except for the save which can be done in the cleanup